File: 3788-maximum-unique-subarray-sum-after-deletion/maximum-unique-subarray-sum-after-deletion.cpp
class Solution {
public:
    int maxSum(vector<int>& nums) {
        set<int> st;
        int maxsum = 0;

        for (auto num : nums) {
            if (num > 0 && st.find(num) == st.end()) {
                maxsum += num;
            }
            st.insert(num);
        }

        return maxsum > 0 ? maxsum : *max_element(nums.begin(), nums.end());
    }
};

File: 3788-maximum-unique-subarray-sum-after-deletion/maximum-unique-subarray-sum-after-deletion.py
class Solution:
    def maxSum(self, nums: List[int]) -> int:
        st = set()
        maxsum = 0

        for num in nums:
            if num not in st and num > 0:
                maxsum += num
            st.add(num)

        return maxsum if maxsum > 0 else max(nums)
